src/components/atoms/Heading/Heading.tsx

```diff
@@ -1,77 +1,83 @@
+import React, { FC, HTMLAttributes } from 'react';
 import { Colors } from '../../../constants/colors';
+import { useThemeContext, AppThemeProps, AppTheme } from '../../styles/Theme';
 
 export type HeadingTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';
 
-export interface StyledHeadingProps extends HTMLAttributes<HTMLHeadingElement> {
-  /**
-   * The HTML5 tag you want to render your heading, it's used to determine the size of the heading as well.
-   */
-  as: HeadingTags;
+export interface OptionalHeadingProps extends HTMLAttributes<HTMLHeadingElement> {
   /**
    * Override the default size assigned to the rendered HTML tag.
    * @default `as`
    */
-  size?: keyof typeof typography.sizes.heading;
+  size?: keyof AppTheme['typography']['heading']['sizes'];
   /**
-   * Accepts a subset of the Zopa brand colors. Same as the ones accepted by `<Text />`.
+   * Accepts a subset of the Zopa brand colors by default. Same as the ones accepted by `<Text />`.
    * @default `colors.greyDarkest`
    */
-  color?: Colors['white'] | Colors['grey'] | Colors['greyDarkest'];
+  color?: Colors['white'] | Colors['grey'] | Colors['greyDarkest'] | string;
   /**
    * Where the rendered text should be aligned to.
    * @default 'inherit'
    */
   align?: 'inherit' | 'left' | 'right' | 'center';
 }
 
-const {
-  sizes: { heading: headingSizes, lineHeight: lineHeightMap },
-} = typography;
+interface HeadingPropsWithAs extends OptionalHeadingProps {
+  /**
+   * The HTML5 tag you want to render your heading, it's used to determine the size of the heading as well.
+   * @default 'required if 'forwardedAs' is not provided.'
+   */
+  as: HeadingTags;
+}
+
+interface HeadingPropsWithForwardedAs extends OptionalHeadingProps {
+  /**
+   * The HTML5 tag you want to render your heading, it's used to determine the size of the heading as well.
+   * To be used if Component has been wrapped in styled components.
+   * @default 'required if 'as' is not provided.'
+   */
+  forwardedAs: HeadingTags;
+}
 
-const letterSpacingMap = {
-  display: '-2.86px',
-  h1: '-1.25px',
-  h2: '-0.85px',
-  h3: '-0.45px',
-  h4: '-0.25px',
-  h5: '-0.02px',
-  h6: '-0.01px',
-};
+export type HeadingProps = HeadingPropsWithAs | HeadingPropsWithForwardedAs;
 
-const Heading = styled.h1<StyledHeadingProps>`
-  ${({ as, size }) => {
-    const tag = size || (as === 'span' ? 'h4' : as);
-    return css`
-      font-size: ${headingSizes[tag]};
-      line-height: ${lineHeightMap[tag]};
-      letter-spacing: ${letterSpacingMap[tag]};
-    `;
+const StyledHeading = styled.h1<HeadingPropsWithAs & AppThemeProps>`
+  ${({ as, size, theme }) => {
+    if (as) {
+      const tag = size || (as === 'span' ? 'h4' : as);
+      return css`
+        font-size: ${theme.typography.heading.sizes[tag]};
+        line-height: ${theme.typography.lineHeight[tag]};
+        letter-spacing: ${theme.typography.letterSpacingMap[tag]};
+      `;
+    }
```

Comment:
Do we want to maybe log a warning or something if for whatever reason `as` isn't passed?

Reply:
Would it not be caught by type checking?

Reply:
Then is the `if (as)` redundant?

Reply:
TypeScript thinks the `as` is nullable

Reply:
Actually you are right, if a user directly passes forwardAs to <Heading/> as will be forwarded to an element which doesn't exist and as wouldn't exist on Heading itself. I guess a warning would make sense :)